schema.ts

```diff
@@ -392,6 +397,84 @@ export interface ScaleSchema extends BaseNumberSchema<ValueHintType.Scale> {
   icon?: ScaleIconSet;
 }
 
+/**
+ * Display types that can be used with an {@link EmailSchema}.
+ */
+export enum EmailDisplayType {
+  /**
+   * Display both icon and email (default).
+   */
+  IconAndEmail = 'iconAndEmail',
+
+  /**
+   * Display icon only.
+   */
+  IconOnly = 'iconOnly',
+
+  /**
+   * Display email address only.
+   */
+  EmailOnly = 'emailOnly',
+}
+
+/**
+ * A schema representing a return value or object property that is an email address.
+ *
+ * An email address can be represented visually as an icon, an icon plus the email address, or
+ * the just the email address.  Auto-complete against previously typed domain names is
+ * also an option in the user interface.
+ */
+export interface EmailSchema extends BaseStringSchema<ValueHintType.Email> {
+  /** Instructs Coda to render this value as a scale. */
+  codaType: ValueHintType.Email;
+  /** How the email should be displayed in the UI. */
+  display?: EmailDisplayType;
+  /** Whether to auto-complete the email domain during user input. */
+  autocomplete?: boolean;
+}
+
+/**
+ * Display types that can be used with a {@link LinkSchema}.
+ */
+export enum LinkDisplayType {
+  /**
+   * Display icon only.
+   */
+  IconOnly = 'IconOnly',
```

Comment:
These values should probably match the `coda` values exactly (which I think use `iconOnly`, `url`, etc).

Reply:
Thank you for the nudge to verify.  I believe in this case they are capitalized, though.  Declarations [here](https://github.com/coda/coda/blob/8cb3fa17d3605a000c98f4429501e4c3bcb3bc74/modules/common/serialized-types/formats.ts#L91).

Reply:
Apologies for the inconsistency - endeavoring to follow "when in Rome" in both places, but the Romans unfortunately don't agree in this case.  Will make the schema match our API.

Does the [x-tsEnumNames here](https://github.com/coda/coda/blob/d55a47d9efd4f9a6394c62e8fa87eb56fce9d176/modules/server/api/oas/v1/schemas/columns.yaml#L236) accomplish the mapping, or must it be done elsewhere?

Reply:
The mapping from packs SDK values back to Coda internal values happens in this file, you can see where it assumes the enum values are the same by using castEnum() in a few places:

https://github.com/coda/coda/blob/main/modules/common/packs-helpers/index.ts#L340

So the new switch case for the new ValueHintType would need to explicitly map each enum value instead.

(x-tsEnumNames is a little thing that just generates the TypeScript enum name in generated code, the `Foo` in `Foo = 'bar'` in the enum declaration)